d2layouts/d2sequence/layout.go

```diff
@@ -4,90 +4,135 @@ import (
 	"context"
 	"fmt"
 	"math"
+	"sort"
 
 	"oss.terrastruct.com/d2/d2graph"
 	"oss.terrastruct.com/d2/lib/geo"
 	"oss.terrastruct.com/d2/lib/go2"
 	"oss.terrastruct.com/d2/lib/label"
+	"oss.terrastruct.com/d2/lib/shape"
 )
 
 func Layout(ctx context.Context, g *d2graph.Graph) (err error) {
-	edgeYStep := MIN_EDGE_DISTANCE
-	actorXStep := MIN_ACTOR_DISTANCE
-	maxActorHeight := 0.
+	sd := &sequenceDiagram{
+		graph:          g,
+		objectRank:     make(map[*d2graph.Object]int),
+		objectDepth:    make(map[*d2graph.Object]int),
+		minEdgeRank:    make(map[*d2graph.Object]int),
+		maxEdgeRank:    make(map[*d2graph.Object]int),
+		edgeYStep:      MIN_EDGE_DISTANCE,
+		actorXStep:     MIN_ACTOR_DISTANCE,
+		maxActorHeight: 0.,
+	}
+
+	sd.init()
```

Comment:
Is there a reason we want this split setup over something like `sd := NewSequenceDiagram(g)` instead? will these values be configured regularly? the maps seem like they can be in `init` unless there is a scenario where we want to preload them with values before calling `init`.

Reply:
I just decided there was no need for `NewSequenceDiagram` right now as it is only used in `Layout`.
Maybe, later on when I work on having them as parts in bigger diagrams I 'll put this in another function to have proper initialization